codeworld-compiler test suite is failing

I'm getting 100% failures from the codeworld-compiler test suite, running on the CodeWorld server.  They all seem to related to one of two things:

1. Whether there are one or two newlines at the end of the line.
2. Unicode issues: e.g., \8226 is turned into \226\8364\162, which looks like a UTF-8 expansion.

```
### Failure in: 0:test_pointType           
test/Main.hs:69
test_pointType
expected: "\nprogram.hs:7:7: error:\n    \8226 Couldn't match type [Number] with (Number, Number)\n      Expected type: Point\n        Actual type: [Number]\n    \8226 In the expression: [2, - 4]\n      In an equation for end: end = [2, - 4]\n"
 but got: "\nprogram.hs:7:7: error:\n    \226\8364\162 Couldn't match type [Number] with (Number, Number)\n      Expected type: Point\n        Actual type: [Number]\n    \226\8364\162 In the expression: [2, - 4]\n      In an equation for end: end = [2, - 4]\n\n"
### Failure in: 1:test_colorName           
test/Main.hs:69
test_colorName
expected: "\nprogram.hs:2:27: error:\n    \8226 Variable not in scope: reds :: Color\n    \8226 Perhaps you meant red (defined in the standard library)\n"
 but got: "\nprogram.hs:2:27: error:\n    \226\8364\162 Variable not in scope: reds :: Color\n    \226\8364\162 Perhaps you meant red (defined in the standard library)\n\n"
### Failure in: 2:test_nesting             
test/Main.hs:69
test_nesting
expected: "\nprogram.hs:1:18: error:\n    \8226 Variable not in scope: diamond :: Picture\n    \8226 Perhaps you meant diamonds (line 2)\n"
 but got: "\nprogram.hs:1:18: error:\n    \226\8364\162 Variable not in scope: diamond :: Picture\n    \226\8364\162 Perhaps you meant diamonds (line 2)\n\n"
### Failure in: 3:test_animation           
test/Main.hs:69
test_animation
expected: "\nprogram.hs:1:28: error:\n    \8226 Couldn't match expected type Number -> Picture\n                  with actual type Picture\n    \8226 In the parameter(s) of animationOf, namely (propellor)\n      In the expression: animationOf (propellor)\n      In an equation for main: main = animationOf (propellor)\n"
 but got: "\nprogram.hs:1:28: error:\n    \226\8364\162 Couldn't match expected type Number -> Picture\n                  with actual type Picture\n    \226\8364\162 In the parameter(s) of animationOf, namely (propellor)\n      In the expression: animationOf (propellor)\n      In an equation for main: main = animationOf (propellor)\n\n"
### Failure in: 4:test_combiningAnimation  
test/Main.hs:69
test_combiningAnimation
expected: "\nprogram.hs:1:20: error:\n    \8226 Couldn't match expected type Picture\n                  with actual type Number -> Picture\n    \8226 Probable cause: a is applied to too few arguments\n      In the parameter(s) of (&), namely a\n      In the parameter(s) of animationOf, namely (a & b)\n      In the expression: animationOf (a & b)\n\nprogram.hs:1:20: error:\n    \8226 Couldn't match expected type Number -> Picture\n                  with actual type Picture\n    \8226 Possible cause: (&) is applied to too many arguments\n      In the parameter(s) of animationOf, namely (a & b)\n      In the expression: animationOf (a & b)\n      In an equation for main: main = animationOf (a & b)\n\nprogram.hs:1:24: error:\n    \8226 Couldn't match expected type Picture\n                  with actual type Number -> Picture\n    \8226 Probable cause: b is applied to too few arguments\n      In the second argument of (&), namely b\n      In the parameter(s) of animationOf, namely (a & b)\n      In the expression: animationOf (a & b)\n"
 but got: "\nprogram.hs:1:20: error:\n    \226\8364\162 Couldn't match expected type Picture\n                  with actual type Number -> Picture\n    \226\8364\162 Probable cause: a is applied to too few arguments\n      In the parameter(s) of (&), namely a\n      In the parameter(s) of animationOf, namely (a & b)\n      In the expression: animationOf (a & b)\n\nprogram.hs:1:20: error:\n    \226\8364\162 Couldn't match expected type Number -> Picture\n                  with actual type Picture\n    \226\8364\162 Possible cause: (&) is applied to too many arguments\n      In the parameter(s) of animationOf, namely (a & b)\n      In the expression: animationOf (a & b)\n      In an equation for main: main = animationOf (a & b)\n\nprogram.hs:1:24: error:\n    \226\8364\162 Couldn't match expected type Picture\n                  with actual type Number -> Picture\n    \226\8364\162 Probable cause: b is applied to too few arguments\n      In the second argument of (&), namely b\n      In the parameter(s) of animationOf, namely (a & b)\n      In the expression: animationOf (a & b)\n\n"
### Failure in: 5:test_simulation          
test/Main.hs:69
test_simulation
expected: "\nprogram.hs:1:32: error:\n    \8226 Couldn't match expected type ([Number] -> world0,\n                                     (world0, Number) -> world0, world0 -> Picture)\n                  with actual type (((Number, Number), Number) -> (Number, Number),\n                                     (Number, Number) -> Picture)\n    \8226 In the parameter(s) of simulationOf, namely (step, picture)\n      In the expression: simulationOf (step, picture)\n      In an equation for main: main = simulationOf (step, picture)\n"
 but got: "\nprogram.hs:1:32: error:\n    \226\8364\162 Couldn't match expected type ([Number] -> world0,\n                                     (world0, Number) -> world0, world0 -> Picture)\n                  with actual type (((Number, Number), Number) -> (Number, Number),\n                                     (Number, Number) -> Picture)\n    \226\8364\162 In the parameter(s) of simulationOf, namely (step, picture)\n      In the expression: simulationOf (step, picture)\n      In an equation for main: main = simulationOf (step, picture)\n\n"
### Failure in: 6:test_missingBracket      
test/Main.hs:69
test_missingBracket
expected: "\nprogram.hs:2:1: error:\n    parse error (possibly incorrect indentation or mismatched brackets)\n"
 but got: "\nprogram.hs:1:27: error:\n    parse error (possibly incorrect indentation or mismatched brackets)\n\n"
### Failure in: 7:test_ifCondition         
test/Main.hs:69
test_ifCondition
expected: "\nprogram.hs:2:37: error: parse error on input else\n"
 but got: "\nprogram.hs:2:37: error: parse error on input else\n\n"
### Failure in: 8:test_arguments           
test/Main.hs:69
test_arguments
expected: "\nprogram.hs:1:27: error:\n    \8226 Couldn't match expected type (Number, Number)\n                  with actual type (Number, Number, Number, Number)\n    \8226 In the parameter(s) of rectangle, namely (2, 2, 2, 2)\n      In the parameter(s) of drawingOf, namely\n        (rectangle (2, 2, 2, 2))\n      In the expression: drawingOf (rectangle (2, 2, 2, 2))\n"
 but got: "\nprogram.hs:1:27: error:\n    \226\8364\162 Couldn't match expected type (Number, Number)\n                  with actual type (Number, Number, Number, Number)\n    \226\8364\162 In the parameter(s) of rectangle, namely (2, 2, 2, 2)\n      In the parameter(s) of drawingOf, namely\n        (rectangle (2, 2, 2, 2))\n      In the expression: drawingOf (rectangle (2, 2, 2, 2))\n\n"
### Failure in: 9:test_listComprehension   
test/Main.hs:69
test_listComprehension
expected: "\nprogram.hs:2:35: error:\n    parse error on input =\n    Perhaps you need a 'let' in a 'do' block?\n    e.g. 'let x = 5' instead of 'x = 5'\n"
 but got: "\nprogram.hs:2:35: error:\n    parse error on input =\n    Perhaps you need a 'let' in a 'do' block?\n    e.g. 'let x = 5' instead of 'x = 5'\n\n"
### Failure in: 10:test_variableName         
test/Main.hs:69
test_variableName
expected: "\nprogram.hs:1:18: error:\n    \8226 Variable not in scope: circl :: Picture\n    \8226 Perhaps you meant one of these:\n        circ (line 2), circle (defined in the standard library)\n"
 but got: "\nprogram.hs:1:18: error:\n    \226\8364\162 Variable not in scope: circl :: Picture\n    \226\8364\162 Perhaps you meant one of these:\n        circ (line 2), circle (defined in the standard library)\n"
### Failure in: 11:test_numberType           
test/Main.hs:69
test_numberType
expected: "\nprogram.hs:5:21: error:\n    \8226 Couldn't match expected type Number with actual type Picture\n    \8226 In the parameter(s) of solidCircle, namely (size)\n      In the expression: solidCircle (size)\n      In an equation for wheel: wheel = solidCircle (size)\n\nprogram.hs:8:8: error:\n    \8226 Couldn't match expected type Picture with actual type Number\n    \8226 In the expression: 4\n      In an equation for size: size = 4\n"
 but got: "\nprogram.hs:5:21: error:\n    \226\8364\162 Couldn't match expected type Number with actual type Picture\n    \226\8364\162 In the parameter(s) of solidCircle, namely (size)\n      In the expression: solidCircle (size)\n      In an equation for wheel: wheel = solidCircle (size)\n\nprogram.hs:8:8: error:\n    \226\8364\162 Couldn't match expected type Picture with actual type Number\n    \226\8364\162 In the expression: 4\n      In an equation for size: size = 4\n\n"
### Failure in: 12:test_nakedExpression      
test/Main.hs:69
test_nakedExpression
expected: "\nprogram.hs:1:1: error:\n    Parse error: naked expression at top level\n"
 but got: "\nprogram.hs:1:1: error:\n    Parse error: naked expression at top level\n\n"
Cases: 13  Tried: 13  Errors: 0  Failures: 13
Cases: 13  Tried: 13  Errors: 0  Failures: 13
Test suite codeworld-compiler-test: FAIL
Test suite logged to:
dist/test/codeworld-compiler-0.1.0.0-codeworld-compiler-test.log
0 of 1 test suites (0 of 1 test cases) passed.
```

--- a/codeworld-compiler/test/testcase/test_animation/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_animation/saved_error.txt
@@ -1,8 +1,7 @@
 
 program.hs:1:28: error:
-    â€¢ Couldn't match expected type Number -> Picture
+    \8226 Couldn't match expected type Number -> Picture
                   with actual type Picture
-    â€¢ In the parameter(s) of animationOf, namely (propellor)
+    \8226 In the parameter(s) of animationOf, namely (propellor)
       In the expression: animationOf (propellor)
       In an equation for main: main = animationOf (propellor)
-

--- a/codeworld-compiler/test/testcase/test_animation/source.hs
+++ b/codeworld-compiler/test/testcase/test_animation/source.hs
@@ -1,3 +1,2 @@
 main         = animationOf(propellor)
 propellor = rotated(solidRectangle(10, 1), 60)
-

--- a/codeworld-compiler/test/testcase/test_arguments/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_arguments/saved_error.txt
@@ -1,9 +1,8 @@
 
 program.hs:1:27: error:
-    â€¢ Couldn't match expected type (Number, Number)
+    \8226 Couldn't match expected type (Number, Number)
                   with actual type (Number, Number, Number, Number)
-    â€¢ In the parameter(s) of rectangle, namely (2, 2, 2, 2)
+    \8226 In the parameter(s) of rectangle, namely (2, 2, 2, 2)
       In the parameter(s) of drawingOf, namely
         (rectangle (2, 2, 2, 2))
       In the expression: drawingOf (rectangle (2, 2, 2, 2))
-

--- a/codeworld-compiler/test/testcase/test_colorName/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_colorName/saved_error.txt
@@ -1,5 +1,4 @@
 
 program.hs:2:27: error:
-    â€¢ Variable not in scope: reds :: Color
-    â€¢ Perhaps you meant red (defined in the standard library)
-
+    \8226 Variable not in scope: reds :: Color
+    \8226 Perhaps you meant red (defined in the standard library)

--- a/codeworld-compiler/test/testcase/test_colorName/source.hs
+++ b/codeworld-compiler/test/testcase/test_colorName/source.hs
@@ -1,4 +1,3 @@
 main     = drawingOf(redWheel)
 redWheel = colored(wheel, reds)
 wheel    = solidCircle(4)
-

--- a/codeworld-compiler/test/testcase/test_combiningAnimation/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_combiningAnimation/saved_error.txt
@@ -1,25 +1,24 @@
 
 program.hs:1:20: error:
-    â€¢ Couldn't match expected type Picture
+    \8226 Couldn't match expected type Picture
                   with actual type Number -> Picture
-    â€¢ Probable cause: a is applied to too few arguments
+    \8226 Probable cause: a is applied to too few arguments
       In the parameter(s) of (&), namely a
       In the parameter(s) of animationOf, namely (a & b)
       In the expression: animationOf (a & b)
 
 program.hs:1:20: error:
-    â€¢ Couldn't match expected type Number -> Picture
+    \8226 Couldn't match expected type Number -> Picture
                   with actual type Picture
-    â€¢ Possible cause: (&) is applied to too many arguments
+    \8226 Possible cause: (&) is applied to too many arguments
       In the parameter(s) of animationOf, namely (a & b)
       In the expression: animationOf (a & b)
       In an equation for main: main = animationOf (a & b)
 
 program.hs:1:24: error:
-    â€¢ Couldn't match expected type Picture
+    \8226 Couldn't match expected type Picture
                   with actual type Number -> Picture
-    â€¢ Probable cause: b is applied to too few arguments
+    \8226 Probable cause: b is applied to too few arguments
       In the second argument of (&), namely b
       In the parameter(s) of animationOf, namely (a & b)
       In the expression: animationOf (a & b)
-

--- a/codeworld-compiler/test/testcase/test_combiningAnimation/source.hs
+++ b/codeworld-compiler/test/testcase/test_combiningAnimation/source.hs
@@ -1,4 +1,3 @@
 main = animationOf(a & b)
 a(t) = rotated(solidRectangle(1, 1), 45 * t)
 b(t) = circle(t)
-

--- a/codeworld-compiler/test/testcase/test_ifCondition/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_ifCondition/saved_error.txt
@@ -1,3 +1,2 @@
 
 program.hs:2:37: error: parse error on input else
-

--- a/codeworld-compiler/test/testcase/test_ifCondition/source.hs
+++ b/codeworld-compiler/test/testcase/test_ifCondition/source.hs
@@ -1,3 +1,2 @@
 main = drawingOf(thing(1) & thing(2))
 thing(n) = if n > 1 rectangle(n, n) else circle(n)
-

--- a/codeworld-compiler/test/testcase/test_listComprehension/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_listComprehension/saved_error.txt
@@ -3,4 +3,3 @@ program.hs:2:35: error:
     parse error on input =
     Perhaps you need a 'let' in a 'do' block?
     e.g. 'let x = 5' instead of 'x = 5'
-

--- a/codeworld-compiler/test/testcase/test_listComprehension/source.hs
+++ b/codeworld-compiler/test/testcase/test_listComprehension/source.hs
@@ -1,3 +1,2 @@
 main = drawingOf(target)
 target = pictures([ circle(r) | r = [1, 2, 3, 4, 5] ])
-

--- a/codeworld-compiler/test/testcase/test_missingBracket/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_missingBracket/saved_error.txt
@@ -1,4 +1,3 @@
 
 program.hs:1:27: error:
     parse error (possibly incorrect indentation or mismatched brackets)
-

--- a/codeworld-compiler/test/testcase/test_nakedExpression/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_nakedExpression/saved_error.txt
@@ -1,4 +1,3 @@
 
 program.hs:1:1: error:
     Parse error: naked expression at top level
-

--- a/codeworld-compiler/test/testcase/test_nesting/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_nesting/saved_error.txt
@@ -1,5 +1,4 @@
 
 program.hs:1:18: error:
-    â€¢ Variable not in scope: diamond :: Picture
-    â€¢ Perhaps you meant diamonds (line 2)
-
+    \8226 Variable not in scope: diamond :: Picture
+    \8226 Perhaps you meant diamonds (line 2)

--- a/codeworld-compiler/test/testcase/test_nesting/source.hs
+++ b/codeworld-compiler/test/testcase/test_nesting/source.hs
@@ -1,3 +1,2 @@
 main = drawingOf(diamond)
 diamonds = rotated(rectangle(2, 2), 45)
-

--- a/codeworld-compiler/test/testcase/test_numberType/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_numberType/saved_error.txt
@@ -1,12 +1,11 @@
 
 program.hs:5:21: error:
-    â€¢ Couldn't match expected type Number with actual type Picture
-    â€¢ In the parameter(s) of solidCircle, namely (size)
+    \8226 Couldn't match expected type Number with actual type Picture
+    \8226 In the parameter(s) of solidCircle, namely (size)
       In the expression: solidCircle (size)
       In an equation for wheel: wheel = solidCircle (size)
 
 program.hs:8:8: error:
-    â€¢ Couldn't match expected type Picture with actual type Number
-    â€¢ In the expression: 4
+    \8226 Couldn't match expected type Picture with actual type Number
+    \8226 In the expression: 4
       In an equation for size: size = 4
-

--- a/codeworld-compiler/test/testcase/test_numberType/source.hs
+++ b/codeworld-compiler/test/testcase/test_numberType/source.hs
@@ -6,4 +6,3 @@ wheel = solidCircle(size)
 
 size :: Picture
 size = 4
-

--- a/codeworld-compiler/test/testcase/test_pointType/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_pointType/saved_error.txt
@@ -1,8 +1,7 @@
 
 program.hs:7:7: error:
-    â€¢ Couldn't match type [Number] with (Number, Number)
+    \8226 Couldn't match type [Number] with (Number, Number)
       Expected type: Point
         Actual type: [Number]
-    â€¢ In the expression: [2, - 4]
+    \8226 In the expression: [2, - 4]
       In an equation for end: end = [2, - 4]
-

--- a/codeworld-compiler/test/testcase/test_pointType/source.hs
+++ b/codeworld-compiler/test/testcase/test_pointType/source.hs
@@ -5,4 +5,3 @@ start = (0, 0)
 
 end :: Point
 end = [2, -4]
-

--- a/codeworld-compiler/test/testcase/test_simulation/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_simulation/saved_error.txt
@@ -1,10 +1,9 @@
 
 program.hs:1:32: error:
-    â€¢ Couldn't match expected type ([Number] -> world0,
+    \8226 Couldn't match expected type ([Number] -> world0,
                                      (world0, Number) -> world0, world0 -> Picture)
                   with actual type (((Number, Number), Number) -> (Number, Number),
                                      (Number, Number) -> Picture)
-    â€¢ In the parameter(s) of simulationOf, namely (step, picture)
+    \8226 In the parameter(s) of simulationOf, namely (step, picture)
       In the expression: simulationOf (step, picture)
       In an equation for main: main = simulationOf (step, picture)
-

--- a/codeworld-compiler/test/testcase/test_simulation/source.hs
+++ b/codeworld-compiler/test/testcase/test_simulation/source.hs
@@ -1,4 +1,3 @@
 main             = simulationOf(step, picture)
 step((x, y), dt) = (x - y * dt, y + x * dt)
 picture(x, y)    = translated(rectangle(1, 1), x, y)
-

--- a/codeworld-compiler/test/testcase/test_variableName/saved_error.txt
+++ b/codeworld-compiler/test/testcase/test_variableName/saved_error.txt
@@ -1,5 +1,5 @@
 
 program.hs:1:18: error:
-    â€¢ Variable not in scope: circl :: Picture
-    â€¢ Perhaps you meant one of these:
+    \8226 Variable not in scope: circl :: Picture
+    \8226 Perhaps you meant one of these:
         circ (line 2), circle (defined in the standard library)

